main.cpp

```diff
@@ -122,394 +172,476 @@ struct CarWash
+{
+    return time * 0.1;
+}
+
+void Cat::eat(char foodType)
+{
+    if (foodType == 'F')
+    {
+        Cat::mew(3);
+    }
+}
+
+void Cat::sleep (float time)
+{
+    time--;
+}
+
+void Cat::mew (int count)
+{
+    count--;
+}
+
+void SpaceShip::CrewMember::examineAnimal(int date, float time, Cat cat)
+{
+    int testNum = 0;
+    if (cat.liveNumber <= 9)
+    {
+        recordTest(date, time, testNum++);
+    }
+    
+    date++;
+    time++;
+}
+
+bool SpaceShip::CrewMember::recordTest(int date, float time, int testNum)
+{
+    date++;
+    time++;
+    
+    if (testNum !=0)
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+}
+
+void SpaceShip::CrewMember::examineCrew (int date, float time, CrewMember id)
+{
+    int testNum = 0;
+    if (id.weight >= 70.0f)
+    {
+        recordTest(date, time, testNum++);
+        date++;
+        time++;
+    }
+}
+
+bool SpaceShip::dock()
+{
+    SpaceShip Rassvet;
+    if (Rassvet.orbitHeight != 40.0f)
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+}
+void SpaceShip::makeLoop(int planetNum )
+{
+    planetNum++;
+}
+
+bool SpaceShip::takeOf(float startTime)
+{
+    SpaceShip Rassvet;
+    startTime--;
+    if (Rassvet.orbitHeight != 0.0f)
+    {
+        return true;
+    } 
+    else
+    {
+        return false;
+    }
+}
+
+void DAC::readInput(int channelNum)
+{
+    for (int i=0;i<channelNum;i++)
+    {
+        checkError(i);
+    }
+}
+
+bool DAC::checkError(double sampleNum)
+{
+    double originalValue = 3.141592;
+    if (sampleNum > originalValue || sampleNum < originalValue)
+    {
+        return false;
+    }
+    else
+    {
+        return true;
+    }
+}
+
+void DAC::audioOut(int channelNum)
+{
+    HeadphoneAmp ampLeft, ampRight;
+    
+    //int bits = 24;
+    
+    for (int i=0;i<channelNum;i++)
+    {
+        ampLeft.doAmp(i);
+        ampRight.doAmp(i);
+    
+    }
+}
+
+bool PowerUnit::getElectricity(int outletStandart)
+{
+    if (outletStandart<3) return true;
+    return false;
+}
+
+double PowerUnit::convertVoltage(double in, double out)
+{
+return std::abs(in - out);
+}
+
+bool PowerUnit::status(int circuitId)
+{
+    if (circuitId == 0)
+    {
+        return false;
+    }
+    else
+    {
+        return true;
+    }
+}
+
+void VCA::attenuate(int coefficient)
+{
+    int knobValue = readKnob();
+    knobValue = knobValue * coefficient;
+}
+
+void VCA::inputPower(int amountOfPower)
+{
+    PowerUnit PSU0;
+    if (amountOfPower < 20)
+    {
+        PSU0.getElectricity(2);
+    }
+}
+
+int VCA::readKnob (int knobAngle)
+{
+    int gain = knobAngle * 1000;
+    return gain;
+}
+
+void HeadphoneAmp::getInput(int channelNum)
+{
+    DAC DACLeft, DACRight;
+    for (int i=0;i<channelNum;i++)
+    {
+        DACLeft.readInput(i);
+        DACRight.readInput(i);
+    }
+
+}
+
+void HeadphoneAmp::doAmp(int channelNum)
+{
+    VCA VCA0;
+    //int output;
+    for (int i=0;i<channelNum;i++)
```

Comment:
add space around your operators to improve the readability of your code.
- [ ] revise